src/api/services/auth.ts

```diff
@@ -0,0 +1,118 @@
+import axios from 'axios';
+import api from '../api';
+import { SafeUser, UserLogInInput, UserSignUpInput } from 'models/user.model';
+
+export interface MeResponse {
+  message: string;
+  user: SafeUser;
+}
+
+type LoginErrorDetails = {
+  formErrors: string[];
+  fieldErrors: {
+    [key: string]: string[];
+  };
+};
+
+export type LoginResult = {
+  token: string | null;
+  errors: LoginErrorDetails | null;
+};
+
+type SignupSuccessResponse = {
+  acknowledged: boolean;
+  insertedId: string;
+};
```

Comment:
`MeResponse` and `SignupSuccessResponse` should ideally be in shared as backend expects to return same type structure, this could be done in a separate PR.

Something like this for signup, as it can return either success or error response:
```
export type SignupResponse = SignupSuccessResponse | SignupErrorResponse;
```